data/publisher/src/main/java/tech/pegasys/teku/data/publisher/MetricsPublisherManager.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.data.publisher;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.io.IOException;
+import java.time.Duration;
+import okhttp3.OkHttpClient;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import tech.pegasys.teku.infrastructure.async.AsyncRunner;
+import tech.pegasys.teku.infrastructure.async.AsyncRunnerFactory;
+import tech.pegasys.teku.infrastructure.metrics.MetricsEndpoint;
+import tech.pegasys.teku.provider.JsonProvider;
+
+public class MetricsPublisherManager {
+
+  private static final long intervalBetweenPublications = 60;
+
+  private final AsyncRunnerFactory asyncRunnerFactory;
+  private final MetricsEndpoint metricsConfig;
+  private static final Logger LOG = LogManager.getLogger();
+  private final JsonProvider jsonProvider = new JsonProvider();
+  private final MetricsDataFactory dataFactory;
+
+  public MetricsPublisherManager(
+      AsyncRunnerFactory asyncRunnerFactory, final MetricsEndpoint metricsConfig) {
+    this.asyncRunnerFactory = asyncRunnerFactory;
+    this.metricsConfig = metricsConfig;
+    this.dataFactory = new MetricsDataFactory(metricsConfig.getMetricsSystem());
+  }
+
+  public void runPublisher() {
```

Comment:
nit: I'd call this `start` for consistency with other classes.

And actually probably worth extending `Service` so it can handle ensuring the service is only started once etc.  You can implement the `doStop()` it requires by storing the `Cancellable` instance that `runWithFixedDleay` will return then cancelling it on stop.